pwa/src/components/object_entities/dataForm.tsx

```diff
@@ -136,44 +80,59 @@ export const ObjectEntityForm:React.FC<ObjectEntityFormProps> = ({ id, entityId
     body = removeEmptyObjectValues(body);
 
     if (!checkValues([body["uri"]])) {
+      setAlert(null);
+      setAlert({type: 'danger', message: 'Required fields are empty'});
+      setShowSpinner(false);
+      return;
+    } else if (!validator.isURL(body["uri"])) {
+      setAlert(null);
+      setAlert({type: 'danger', message: 'Uri is invalid'});
+      setShowSpinner(false);
       return;
     }
 
-    let url = `${context.adminUrl}/object_entities`;
-    let method = "POST";
-    if (id !== "new") {
-      url = `${url}/${id}`;
-      method = "PUT";
+    if (!entity_objectId) { // unset id means we're creating a new entry
+      API.Entity_objects.create(body)
+        .then((res) => {
+          setEntity_object(res.data)
```

Comment:
When we set the `state` here, but on the next line we `navigate` away. What does updating the `state` do?